base/char.jl

```diff
@@ -1,8 +1,24 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
-convert(::Type{Char}, x::UInt32) = reinterpret(Char, x)
+function convert(::Type{UInt32}, c::Char)
+    u = reinterpret(UInt32, c)
+    u ⊻= ifelse(
+        u <= 0x0000ffff,
+        ifelse(u <= 0x000000ff, 0x00000000, 0x0000c080),
+        ifelse(u <= 0x00ffffff, 0x00e08080, 0xf0808080),
+    )
+    ((u & 0x000000ff) >> 0) ⊻ ((u & 0x0000ff00) >> 2) ⊻
+    ((u & 0x00ff0000) >> 4) ⊻ ((u & 0xff000000) >> 6)
+end
+
+function convert(::Type{Char}, u::UInt32)
```

Comment:
Tested and this works OK for all valid Unicode (last code point U+10FFFF). The conversion breaks for invalid Unicode as it overflows at `0x00400000`.
I understand that we want to avoid branching here so adding the test if `u` is not too large is not an option? If so - this probably be documented somewhere.

Reply:
I'm still contemplating how best to handle invalid code points. There was previously an error check here but it caused a massive build time regression (it was also calling a deprecated error type), so I just took it out for now. Will have to figure out what to do here.